src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -313,8 +322,21 @@ dnnl::memory::desc layout_to_memory_desc(cldnn::layout l, dnnl::memory::format_t
     } else if (flatten) {
         dims = flatten_tensor(l.get_tensor());
     } else {
-        auto rank = cldnn::format::dimension(l.format);
-        dims = convert_tensor(l.get_tensor(), rank, cldnn::format::is_grouped(l.format));
+        // clDNN expresses 3d tensor with 4d format. This code is to use 3d format on oneDNN for such case.
+        // However, if the memory::desc to be converted is related to another blocked format, it should be expanded to a 4d tensor.
+        auto shape_rank = l.is_dynamic() ?
+            static_cast<size_t>(l.get_partial_shape().rank().get_length()) : l.get_shape().size();
+        if (shape_rank == 3 && !need_blocked) {
+            dims.push_back(l.batch());
+            dims.push_back(l.feature());
+            // In cldnn::layer, when it is a 3D shape, the values ​​of the XY axes can sometimes be flipped,
+            // so the larger value of the two is used.
+            dims.push_back(std::max(l.spatial(0), l.spatial(1)));
+            target_fmt = dnnl::memory::format_tag::abc;
+        } else {
+            auto rank = cldnn::format::dimension(l.format);
+            dims = convert_tensor(l.get_tensor(), rank, cldnn::format::is_grouped(l.format));
```

Comment:
I'm afraid cached-model failure may happen for blocked format as well? is it OK?